From f90ef4287a689813ac7d151c88978a384abda489 Mon Sep 17 00:00:00 2001
Message-Id: <f90ef4287a689813ac7d151c88978a384abda489.1417465742.git.chang-joon.lee@intel.com>
In-Reply-To: <664d811e547fec51f0453c5acda92012de288ecd.1417465742.git.chang-joon.lee@intel.com>
References: <664d811e547fec51f0453c5acda92012de288ecd.1417465742.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Fri, 14 Nov 2014 14:59:17 +0530
Subject: [PATCH 09/11] FOR_UPSTREAM [VPG]: drm/i915: Support all FB width and
 height from HWC for cases like panel fitter

When Panel fitter is enabled, the fb width and height are lesser than hdisplay/vdisplay.
Avoid -ENOSPC return from drm_crtc_check_viewport in such cases which fails flips.

Issue: GMIN-3045
Change-Id: I9f37cb3664303584e550f8500c7afa0b84ed0ad1
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/drm_crtc.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/drm_crtc.c b/drivers/gpu/drm/drm_crtc.c
index d065e6d..878a44f 100644
--- a/drivers/gpu/drm/drm_crtc.c
+++ b/drivers/gpu/drm/drm_crtc.c
@@ -2466,14 +2466,15 @@ int drm_crtc_check_viewport(const struct drm_crtc *crtc,
 	if (crtc->invert_dimensions)
 		swap(hdisplay, vdisplay);
 
+	/* Support for all width and height for cases like fitter */
 	if (hdisplay > fb->width ||
 	    vdisplay > fb->height ||
 	    x > fb->width - hdisplay ||
 	    y > fb->height - vdisplay) {
-		DRM_DEBUG_KMS("Invalid fb size %ux%u for CRTC viewport %ux%u+%d+%d%s.\n",
-			      fb->width, fb->height, hdisplay, vdisplay, x, y,
-			      crtc->invert_dimensions ? " (inverted)" : "");
-		return -ENOSPC;
+		DRM_DEBUG_KMS(
+			"Invalid fb size %ux%u for CRTC viewport %ux%u+%d+%d%s.\n",
+			fb->width, fb->height, hdisplay, vdisplay, x, y,
+			crtc->invert_dimensions ? " (inverted)" : "");
 	}
 
 	return 0;
-- 
1.7.9.5

